modules/foreman_provisioning_template.py

```diff
@@ -337,13 +346,15 @@ def main():
     state = template_dict.pop('state')
     file_name = template_dict.pop('file_name', None)
 
-    if file_name:
+    if file_name or 'template' in template_dict:
+        if file_name:
+            parsed_dict = parse_template_from_file(file_name, module)
+        else:
+            parsed_dict = parse_template(template_dict['template'], module)
+        # sanitize name from template data
+        if 'name' in parsed_dict and parsed_dict['name'] == '*':
+            module.fail_json(msg="Cannot use '*' as a template name!")
```

Comment:
I believe this is unreachable code. You would have to have `params['name'] == '*' and ('file_name' in params or 'template' in params)`. It would hit [line 332](https://github.com/theforeman/foreman-ansible-modules/pull/19/files#diff-c730a4c9b53728f75758cea520f70012R332).